server/ctrlsubsonic/handlers_raw.go

```diff
@@ -231,12 +232,24 @@ func (c *Controller) ServeStream(w http.ResponseWriter, r *http.Request) *spec.R
 	onCacheHit := func(profile encode.Profile, path string) error {
 		log.Printf("serving transcode `%s`: cache [%s/%dk] hit!\n",
 			audioFile.AudioFilename(), profile.Format, profile.Bitrate)
+		cacheMime, _ := mime.FromExtension(profile.Format)
+		w.Header().Set("Content-Type", cacheMime)
+
+		cacheFile, cfErr := os.Stat(path)
+		if cfErr != nil {
+			log.Printf("failed to stat cache file `%s`: %v", path, cfErr)
+		} else {
+			contentLength := fmt.Sprintf("%d", cacheFile.Size())
+			w.Header().Set("Content-Length", contentLength)
+		}
```

Comment:
i think it's safe to return a real error here because if the stat fails, then the http.ServeFile will fail. thoughts?

```suggestion
		cacheFile, err := os.Stat(path)
		if err != nil {
			return fmt.Errorf("failed to stat cache file `%s`: %v", path, err)
		}
		contentLength := fmt.Sprintf("%d", cacheFile.Size())
		w.Header().Set("Content-Length", contentLength)

```

Reply:
Sounds reasonable, yes. :)